lib/workers/repository/init/index.ts

```diff
@@ -11,21 +11,52 @@ import { mergeRenovateConfig } from './config';
 import { detectSemanticCommits } from './semantic';
 import { detectVulnerabilityAlerts } from './vulnerability';
 
-export async function initRepo(input: RenovateConfig): Promise<RenovateConfig> {
+let cache: repositoryCache.Cache;
+
+function initializeConfig(config: RenovateConfig): RenovateConfig {
+  return { ...config, errors: [], warnings: [], branchList: [] };
+}
+
+async function initializeCaches(config: RenovateConfig): Promise<void> {
   memCache.init();
-  await repositoryCache.initialize(input);
-  let config: RenovateConfig = {
-    ...input,
-    errors: [],
-    warnings: [],
-    branchList: [],
-  };
-  config = await initApis(config);
+  await repositoryCache.initialize(config);
+  cache = repositoryCache.getCache();
+  cache.init = cache.init || {};
+}
+
+function validCache(config: RenovateConfig): boolean {
+  return !!(
+    config.defaultBranch === cache.init.defaultBranch &&
+    cache.init.defaultBranchSha &&
```

Comment:
This only works with github so far, as it returns the default branch Sha from initRepo()

Reply:
for bitbucket server we need to fetch branches endpoint to get current sha 🙃 

https://docs.atlassian.com/bitbucket-server/rest/7.0.1/bitbucket-rest.html#idp203